### PR TITLE
updated docs and fixed file opening

### LINKED ISSUE #42: Broken link in docs
The "using the API" link is broken (404) on https://docs.api.qmk.fm/

It points to https://github.com/qmk/qmk_compiler_api/tree/0debc4a22df4515119ac59ba81aed620d9a10fe1/docs/api_docs.html, maybe it's supposed to be https://github.com/qmk/qmk_api/blob/f381850ccc4cab411614b22808b138544752c48c/docs/api_docs.md?

### PR DESCRIPTION
**Pull request by StrikerRUS**
Fixed #42.

- Update links to use `https`, MinIO spelling;
- markup code snippet;
- provide ref to `setup.cfg` in README;
- close version file after reading.

### CODE REVIEW

File: docs/architecture.svg [hunk 1]
```diff
@@ -1,2 +1,2 @@
...
...90.92 510.64 L 98.67 511.74 L 95.24 513.62 L 94.68 517.5 Z" fill="#6c8ebf" stroke="#6c8ebf" stroke-miterlimit="10" pointer-events="none"/><path d="M 355 600 L 96.03 512.05" fill="none" stroke="#6c8ebf" stroke-miterlimit="10" pointer-events="none"/><path d="M 91.06 510.36 L 98.81 509.3 L 96.03 512.05 L 96.56 515.92 Z" fill="#6c8ebf" stroke="#6c8ebf" stroke-miterlimit="10" pointer-events="none"/><path d="M 370 503 L 393.41 593.83" fill="none" stroke="#d6b656" stroke-miterlimit="10" pointer-events="none"/><path d="M 394.72 598.92 L 389.58 593.01 L 393.41 593.83 L 396.36 591.27 Z" fill="#d6b656" stroke="#d6b656" stroke-miterlimit="10" pointer-events="none"/><path d="M 370 503 L 120.95 597.74" fill="none" stroke="#d6b656" stroke-miterlimit="10" pointer-events="none"/><path d="M 116.04 599.6 L 121.34 593.84 L 120.95 597.74 L 123.83 600.39 Z" fill="#d6b656" stroke="#d6b656" stroke-miterlimit="10" pointer-events="none"/><path d="M 370 503 L 264.78 595.79" fill="none" stroke="#d6b656" stroke-miterlimit="10" pointer-events="none"/><path d="M 260.84 599.26 L 263.77 592.01 L 264.78 595.79 L 268.4 597.26 Z" fill="#d6b656" stroke="#d6b656" stroke-miterlimit="10" pointer-events="none"/></g></svg>
```
Comment (skullydazed) [inline docs/architecture.svg]: You should also update the source file `architecture.draw.io.xml`.

Comment (StrikerRUS) [reply docs/architecture.svg]: Done! Also, during updating sources, I fixed the inconsistency for Redis between source and `svg` files.

**Comment by StrikerRUS** [comment]
gently ping @noroadsleft @skullydazed

**Review by noroadsleft — APPROVED** [review]
The `web.py` changes are slightly over my head (I don't know python), but I don't see anything alarming there. The readme and docs updates all look good to me.

Edit: For the record, I can tell what the `web.py` changes are doing, I just don't have any context for what improvement that actually makes.

**Comment by StrikerRUS** [comment]
@noroadsleft That change just closes the file after reading.

For example, refer to https://stackoverflow.com/a/8011836.

**Comment by noroadsleft** [comment]
Nice, thanks for the link @StrikerRUS.

One of these days I will actually start learning Python.

**Comment by StrikerRUS** [comment]
ping @skullydazed 
Should I do anything else to get this PR merged?

**Review by noroadsleft — APPROVED** [review]

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**